saleor/discount/utils.py

```diff
@@ -345,17 +405,77 @@ def create_or_update_discount_objects_from_promotion_for_checkout(
     lines_info: Iterable["CheckoutLineInfo"],
 ):
     create_discount_objects_for_catalogue_promotions(lines_info)
-    create_discount_objects_for_order_promotions(checkout_info, lines_info)
+    create_checkout_discount_objects_for_order_promotions(checkout_info, lines_info)
+
+
+def create_or_update_discount_objects_from_promotion_for_order(
+    order: "Order",
+    lines_info: Iterable["DraftOrderLineInfo"],
+):
+    create_discount_objects_for_catalogue_promotions(lines_info)
+    _update_order_line_prefetched_discounts(lines_info)
```

Comment:
The `create_discount_objects_for_catalogue_promotions` already updates the `line_info.discounts` so maybe we do not need to call `_update_order_line_prefetched_discounts`

Reply:
You are right, in `discount.utils.py` file we cache discount data in 'lines_info' object. Prefetch is needed in `calculation.py`